pydantic_ai_slim/pydantic_ai/models/__init__.py

```diff
@@ -343,6 +354,22 @@ class ModelRequestParameters:
     output_tools: list[ToolDefinition] = field(default_factory=list)
     allow_text_output: bool = True
 
+    @cached_property
+    def tool_defs(self) -> dict[str, ToolDefinition]:
+        return {tool_def.name: tool_def for tool_def in [*self.function_tools, *self.output_tools]}
+
+    def get_final_result_event(self, e: ModelResponseStreamEvent) -> FinalResultEvent | None:
```

Comment:
Todo: move this to a wrapper of `StreamedResult` that automatically inserts the `FinalResultEvent` and can be used both in `AgentStream.__aiter__` and `TemporalModel`.